### DISCUSSION TITLE
Does performance increase by merging different .csproj projects into 1 big .csproj file

### DISCUSSION THREAD

**Discussion by giannik** [OrchardCMS/OrchardCore#10990] [Q&A]
I have around 30 custom  .csproj projects as modules inside an orchard core solution .
Im looking into merging them all in 1 single .csproj project and seprarating each module using a folder structure with project.
Are there any indicators that this will increase overall performance at build time and at run time once project has been deployed ?
Or is it simply not worth the trouble ?

**Comment by hishamco** [comment]
Projects are way to seperate an organize a BIG project into parts which are physically separated. Folders in other hand are another way to organize but the projects are logically separated

Have a look to OC as example where each module located in separated project

**Reply by giannik** [reply]
Im wondering if there would be a noticable performance benefit from merging all projects into 1 project?

**Comment by Piedone** [comment]
Visual Studio and MSBuild have sophisticated ways to resolve dependencies, parallelize bulids and project loading during development, assembly loading during runtime, as well as deployments. I don't think there is a general answer to this, you'd have to test, but my gut feeling is that even if it's faster it won't make a substantial difference (and it can be slower).

It would be harder to manage during development time though.
